src/network/st_nonblocking/Connection.cpp

```diff
@@ -1,26 +1,173 @@
+                        break;
+                    } 
+                    else {
+                        std::memmove(rbuffer, rbuffer + parsed, readed_bytes - parsed);
+                        readed_bytes -= parsed;
+                    }
+                }
+
+                // There is command, but we still wait for argument to arrive...
+                if (command_to_execute && arg_remains > 0) {
+                    _logger->debug("Fill argument: {} bytes of {}", readed_bytes, arg_remains);
+                    // There is some parsed command, and now we are reading argument
+                    std::size_t to_read = std::min(arg_remains, std::size_t(readed_bytes));
+                    argument_for_command.append(rbuffer, to_read);
+
+                    std::memmove(rbuffer, rbuffer + to_read, readed_bytes - to_read);
+                    arg_remains -= to_read;
+                    readed_bytes -= to_read;
+                }
+
+                // Thre is command & argument - RUN!
+                if (command_to_execute && arg_remains == 0) {
+                    
+                    _logger->debug("Start command execution");
+
+                    std::string result;
+                    command_to_execute->Execute(*pStorage, argument_for_command, result);
+
+                    // Save response
+                    result += "\r\n";
+                    answers.push_back(result);
+
+                    // Prepare for the next command
+                    command_to_execute.reset();
+                    argument_for_command.resize(0);
+                    parser.Reset();
+                }
+            } // while (readed_bytes > 0)
+        } // while(read > 0)
+        if (answers.size() != 0) {
+            _event.events |= EPOLLOUT;
+        }
+        if (readed_bytes == 0) {
+            _logger->debug("Connection closed");
+            OnClose();
+        }
+        else if (errno != EWOULDBLOCK && errno != EAGAIN) {
+            throw std::runtime_error(std::string(strerror(errno)));
+        }
+    }
+    catch (std::runtime_error &ex) {
+        std::string message("SERVER ERROR ");
+        message += ex.what();
+        message += "\r\n";
+        _logger->error("Failed to process connection on descriptor {}: {}", _socket, ex.what());
+        write(_socket, message.data(), message.size());
+        OnError();
+    }
+}
 
 // See Connection.h
-void Connection::DoWrite() { std::cout << "DoWrite" << std::endl; }
+void Connection::DoWrite() {
+    //std::cout << "DoWrite" << std::endl;
+    size_t count = answers.size();
+    iovec tmp[count];
+    //std::unique_ptr<iovec[]> tmp(new iovec[count]);
+    int head_written_count = 0;
+    {
+        std::size_t i = 0;
+        for (auto it = answers.begin(); it != answers.end(); it++) {
+            tmp[i].iov_base = &((*it)[0]);
+            tmp[i].iov_len = it->size();
+            i++;
+        }
+    }
+
+    tmp[0].iov_base += offseto;
```

Comment:
warning:
src/network/st_nonblocking/Connection.cpp:139:21: warning: pointer of type ‘void *’ used in arithmetic [-Wpointer-arith]
     tmp[0].iov_base += offseto;